recipe/meta.yaml

```diff
@@ -29,33 +29,38 @@ outputs:
       run_exports:
         - {{ pin_subpackage('openmpi', min_pin='x.x.x', max_pin='x') }}
       ignore_run_exports:
-        - ucx  # [enable_cuda]
+        - ucx  # [linux]
       script_env:
-        - CUDA_HOME     # [enable_cuda]
+        - CUDA_HOME  # [enable_cuda]
     requirements:
       build:
         - {{ compiler('c') }}
         - {{ compiler('cxx') }}
         - {{ compiler('fortran') }}
-        - autoconf  # [unix]
-        - automake  # [unix]
+        #- autoconf  # [unix]
+        #- automake  # [unix]
         - libtool   # [unix]
         - make  # [unix]
-        - perl 5.26.2
       host:
+        #- openpmix
+        #- prrte
+        - libhwloc
+        - libevent
         - zlib
-        - ucx                 # [enable_cuda]
-        - ucx-proc =*=gpu     # [enable_cuda]
+        - ucx  # [linux]
       run:
-        - zlib
         - mpi 1.0 openmpi
+        #- openpmix
+        #- prrte
+        - libhwloc
+        - libevent
+        - zlib
       run_constrained:
-        - cudatoolkit  >= {{ cudatoolkit }}         # [enable_cuda]
-        - {{ pin_compatible("ucx", max_pin="x") }}  # [enable_cuda]
+        - {{ pin_compatible("ucx", max_pin="x") }}  # [linux]
+        - cudatoolkit  >= {{ cudatoolkit }}  # [enable_cuda]
```

Comment:
Note: On CUDA 12+, we no longer ship `cudatoolkit` that is a repackage of CUDA libraries. It's `cuda-toolkit` which is a metapackage containing all of CUDA Toolkit as conda packages, including compilers etc, but it's an overkill here because Open MPI does not use any (Pending Jeff's reply). We just need to use `cuda-version` mutex to ensure a consistent CUDA environment.
```suggestion
        # Open MPI only uses CUDA Driver APIs, set the minimal driver version
        - __cuda  >= {{ cudatoolkit }}  # [enable_cuda]
        # Ensure a consistent CUDA env
        - cuda-version  >= {{ cudatoolkit }}  # [enable_cuda]
```

Reply:
Then we have to rename `cudatoolkit` -> `cuda_version` in `conda_build_config.yaml`.

Reply:
Yes, it'd be better, it's getting confusing with the transition to new scheme.